### DISCUSSION TITLE
Is this project open source?

### DISCUSSION THREAD

**Discussion by mschipperheyn** [htmlstreamofficial/preline#130] [General]
Hi, I like the look of this library but the license gives me pause. It seems to suggest an open source approach, but it looks a bit thrown together TBH and the reference to "Figma" throws me off. If this library is open source, would it be possible to consider applying an MIT license just to leave any doubts aside? Or is there in fact a commercial license aspect I should consider? If so, what is it? Nothing against commercial. Just want to know what I'm dealing with

**Comment by billturner** [comment]
It is released under the MIT license: https://github.com/htmlstreamofficial/preline/blob/main/LICENSE

**Comment by jahaganiev** [comment]
Hey @mschipperheyn,

Preline UI is fully open source and is licensed under the MIT License, as @billturner mentioned.

Please note that we consider Preline Figma to be a completely separate product category. For this reason, it comes with its own custom license, which allows for free use in both commercial and personal projects.

If you have any concerns or questions, please let us know. We're happy to provide further information.

Cheers!
Jaha

**Comment by chhatrachhorm** [comment]
Hi @jahaganiev, I am sorry for asking this topic again. 

I have a question about the release of newer version 2.5 whose license has been changed from MIT to MIT+Fair Use License.
The fair use license seems to only talk about competing products and derivative products. 
However, if I were to use the preline UI in my commercial website (online learning platform for example), would MIT license apply to mine?
Is it required to release my commercial website under MIT+Fair Use License in this case?

Thanks in advance.

**Reply by jahaganiev** [reply]
Hey @chhatrachhorm - thanks for your question.

No, it's not required to include "Fair Use License" terms in your case since it's online learning platform. It has nothing to do with the distribution of the source code.

Cheers!

**Comment by ssddanbrown** [comment]
Just to add to this, it's not clear if [the license](https://github.com/htmlstreamofficial/preline/blob/0c521a3c454f2ce345cad137473b77a89ed50a33/LICENSE) terms must be taken in combination ("MIT" **and** "Preline UI Fair Use License" must be both adhered to) or whether there's a choice between the two like many dual license scenarios ("MIT" **or** "Preline UI Fair Use License").

If the latter, then the extra terms are fairly redundant since the project can just be used under the MIT. If the latter, then it may be misleading to label the project as MIT or open source since it (when the terms are combined) would not be commonly regarded as either.

**Reply by mohsinhijazee** [reply]
With _Fair Use License_ attached, I think it is more of a _Source available_ then open source because it says _MIT_ and _Fair Use_ both apply. Putting such a library is a labour intensive effort and time consuming effort. I would want the authors to succeed rather burn out in process.

Probably a core could be MIT and commercial license should have advanced components. I am not sure where the line is for one, it is a tricky question to answer almost always and secondly, even if there's a right answer, the authors/creators have the ultimate right and say how they see fit to distribute their software.

Also, I am not clear if a wrapping library (such as Svelte/React components) would be a a violation of Fair Use license or not.

**Reply by ssddanbrown** [reply]
Yeah, I totally respect their right for them to choose a license & distribute how they wish, but just think it could be misleading (as you say, it may be source available rather than open source) & cause problems due to this. As an example, I came across this from another project which was using preline code [as discussed here](https://www.reddit.com/r/opensource/comments/1g0f6fu/comment/lr95vji/), and these terms would have impact to other project using this code.

Interestingly the author there stated:

> We've confirmed with Preline and they have implemented this license just to restrict those who utilize their code without proper attribution. 

But, as reflected in my response to them, that does not really make sense as the "Fair use" terms have stricter requirements than the MIT license.

**Reply by mohsinhijazee** [reply]
But I was under the impression that MIT license alone requires proper attribution as well.

**Reply by ssddanbrown** [reply]
@mohsinhijazee It does, but the attribution requirements for MIT are very basic & easy to achieve (mention of copyright and license), whereas the the "fair use" license terms require a lot more specific forms of attribution, including and in addition to the MIT attribution requirements:

> \- Proper attribution must be given to the original author, including the name "Preline UI" and a link to the original repository.
> \- Redistributions must include this Fair Use License in addition to the MIT License
> \- Redistributions must not remove or alter any licensing information or notices.
> \- Must include clear attribution with a link to the original repository.

**Reply by jahaganiev** [reply]
@ssddanbrown, @mohsinhijazee - thanks for your shared concerns and feedback, we truly appreciate that!

We introduced the “Fair Use License” with certain licensing terms to protect against scenarios that might undermine the project’s sustainability. In most cases, the “Fair Use License” is not even applied.

In the past, we’ve encountered some projects misusing Preline UI by 1:1 replicating it, removing the original copyright notices entirely and labelling it as their own, which violates even the MIT terms. These kinds of situations led us to introduce the additional Fair Use License terms.

While the Fair Use License restricts usage in competing products (such as other component libraries), we are still open to making exceptions if authors commit to adhering to the Fair Use terms and communicate with us the details in advance. The [FlyonUI](https://flyonui.com/) is great example of this, and we truly appreciate FlyonUI’s professionalism and contribution to the ecosystem.

MIT is just one type of the license for open-source projects. There’re large amount of projects in the ecosystem which comes with multiple license terms while being open-source project. While MIT doesn’t require any credits, it does require that original copyright notices remain in the source code files.

Hope this helps clarify any concerns!

Cheers!
Jaha

**Reply by ssddanbrown** [reply]
Thanks for the response @jahaganiev!

> [...] which violates even the MIT terms. These kinds of situations led us to introduce the additional Fair Use License terms.

But if they're already violating your license, they'd also probably be happy to violate those those other terms too?

> There’re large amount of projects in the ecosystem which comes with multiple license terms while being open-source project.

Sure, the issue isn't that you're setting custom/added terms though, it's that:

1. By the license term text alone it's currently unclear if it's a choice between either license option, or if the license is a combination of the MIT and the added terms.
2. The project is marketing as being MIT licensed and open source. Based on your response, I assume you expect the "Preline UI Fair Use License" terms to be respected in combination with the MIT license, in which case the project wouldn't be widely regarded as "MIT licensed" nor "open source", and therefore may be misleading.

Based upon your "Preline UI Fair Use License" and your response, I get the impression your main concern is preventing competitive use rather than meeting attribution requirements, in which case open source in general may not really align with what you want.

**Reply by mohsinhijazee** [reply]
> @ssddanbrown, @mohsinhijazee - thanks for your shared concerns and feedback, we truly appreciate that!

Thank you for a detailed response!

> We introduced the “Fair Use License” with certain licensing terms to protect against scenarios that might undermine the project’s sustainability. In most cases, the “Fair Use License” is not even applied.
> 
> In the past, we’ve encountered some projects misusing Preline UI by 1:1 replicating it, removing the original copyright notices entirely and labelling it as their own, which violates even the MIT terms. These kinds of situations led us to introduce the additional Fair Use License terms.

That's very unfortunate and unfair situation. In my subjective opinion, Preline is the best looking library that's even (my personal opinion) better than the most widely known Tailwind component libraries out there.

> While the Fair Use License restricts usage in competing products (such as other component libraries), we are still open to making exceptions if authors commit to adhering to the Fair Use terms and communicate with us the details in advance. The [FlyonUI](https://flyonui.com/) is great example of this, and we truly appreciate FlyonUI’s professionalism and contribution to the ecosystem.
> 

Off topic - I am wrapping few things as Svelte components and I discover that some of the API can be made more friendly to component use cases where it can be configured without the data attributes on HTML alone and that too without loading the whole library rather leaving the bundler to pick pieces along. I have some raw (possibly incorrect) feedback on that in a PR as well. I feel that the goal of the project seems to be more towards non JS ecosystem (likes of Django, Laravel and anywhere where JS tooling is absent) with a single load of the file at bottom and then instantiation via init methods on document load via attribute querying.

> MIT is just one type of the license for open-source projects. There’re large amount of projects in the ecosystem which comes with multiple license terms while being open-source project. While MIT doesn’t require any credits, it does require that original copyright notices remain in the source code files.

Maybe we can structure/word it differently: Preline is MIT licensed except for direct derivative works that intend to act as component libraries built on top of Preline.

> 
> Hope this helps clarify any concerns!
> 
> Cheers! Jaha

**Reply by jahaganiev** [reply]
Hey everyone - just a quick and a bit delayed update! We've clarified when Preline UI can be used under the MIT license and when the Fair Use License applies. Hope this clears up any questions or concerns.

**Reply by ssddanbrown** [reply]
Thanks @jahaganiev. So going [by the docs](https://preline.co/docs/license.html):

> Preline UI can be used under the MIT License in projects that do not compete with or derive from it as a UI library or similar product. Specifically, the following types of projects would qualify: [...]

I strongly suggest this should also be stated in the license file itself though since it's unclear until you specifically dive into the docs as far as I could tell. 
Also, this is still confusing since it's not being "used under the MIT license" if your extra conditions are to be respected, since the MIT license allows open use and distribution. Ultimately this combination will always be confusing since the licenses are fundamentally conflicting in what they state.

The changes to confirm that you intend the preline fair use terms are to be combined with the MIT, and therefore the end result is not something which would meet [the OSD](https://opensource.org/osd) (or free software definition) and therefore wouldn't be widely considered open source, so it may be misleading to many to call it so. If interested, I've written about why I think using the term properly is important [on my blog here](https://danb.me/blog/why-open-source-term-is-important/).
